packages/pds/src/api/com/atproto/sync/getBlocks.ts

```diff
@@ -5,22 +5,34 @@ import SqlRepoStorage from '../../../../sql-repo-storage'
 import AppContext from '../../../../context'
 import { blocksToCarStream } from '@atproto/repo'
 import { byteIterableToStream } from '@atproto/common'
+import { isUserOrAdmin } from '../../../../auth'
+import { softDeleted } from '../../../../db/util'
 
 export default function (server: Server, ctx: AppContext) {
-  server.com.atproto.sync.getBlocks(async ({ params }) => {
-    const { did } = params
-    const cids = params.cids.map((c) => CID.parse(c))
-    const storage = new SqlRepoStorage(ctx.db, did)
-    const got = await storage.getBlocks(cids)
-    if (got.missing.length > 0) {
-      const missingStr = got.missing.map((c) => c.toString())
-      throw new InvalidRequestError(`Could not find cids: ${missingStr}`)
-    }
-    const car = blocksToCarStream(null, got.blocks)
+  server.com.atproto.sync.getBlocks({
+    auth: ctx.optionalAccessOrAdminVerifier,
+    handler: async ({ params, auth }) => {
+      const { did } = params
+      // takedown check for anyone other than an admin or the user
+      if (!isUserOrAdmin(auth, did)) {
+        const account = await ctx.services.account(ctx.db).getAccount(did, true)
```

Comment:
thoughts on making a simpler method on this service that doesn't need any joins etc?

something like `getAccountState(did): TakenDown | NotExists | Available`

or `isTakenDown(did): boolean`

Reply:
or actually `isRepoAvailable(did): boolean` may be best

Reply:
Just did that quick pass 👍